### PR TITLE
Use non default port 4343 and git user

### PR DESCRIPTION
**Pull request by stuggi**

### CODE REVIEW

File: ansible/git.yaml [hunk 1]
```diff
@@ -3,7 +3,47 @@
...
+    - name: Add TCP firewall rule
+      firewalld:
+        port: "{{ item }}/tcp"
+        state: enabled
+        zone: libvirt
+        permanent: yes
+        immediate: yes
+      with_items:
+        - 4343
+
```
Comment (olliewalsh) [inline ansible/git.yaml:41]: sshd_config says:

```
# If you want to change the port on a SELinux system, you have to tell
# SELinux about this change.
# semanage port -a -t ssh_port_t -p tcp #PORTNUMBER
```

File: ansible/git.yaml [hunk 2]
```diff
@@ -3,7 +3,47 @@
...
   tasks:
+    - name: Remove unwanted ports
+      ansible.builtin.lineinfile:
+        regexp: ^Port (?!{{ ssh_ports | map('regex_escape') | join('|') }}).*$
+        state: absent
+        path: "/etc/ssh/sshd_config"
+
+    - name: Add needed ports to config
+      ansible.builtin.lineinfile:
```
Comment (olliewalsh) [inline ansible/git.yaml:19]: Could maybe set `insertafter: "^#Port"` to ensure it's set at the right place? sshd_config can have Match blocks at the end. Not sure if setting the Port after that will work

File: ansible/git.yaml (hunk 1 above)
Comment (stuggi) [reply ansible/git.yaml:41]: thats what we do in the above task "Allow sshd to listen on tcp port 4343"

**Review by abays — APPROVED** [review]
/lgtm

**Review by olliewalsh — APPROVED** [review]
/lgtm

**Comment by olliewalsh** [comment]
CI failed:
```2023-06-29T16:47:57.800950747Z + git clone ssh://ospdogit@10.9.76.46:4343/home/git/playbooks.git /tmp/tmp.gLkndawumJ
2023-06-29T16:47:57.803268255Z Cloning into '/tmp/tmp.gLkndawumJ'...
2023-06-29T16:47:57.849826142Z Warning: Permanently added '[10.9.76.46]:4343' (ECDSA) to the list of known hosts.
2023-06-29T16:47:58.252781133Z fatal: '/home/git/playbooks.git' does not appear to be a git repository
2023-06-29T16:47:58.252832663Z fatal: Could not read from remote repository.
2023-06-29T16:47:58.252832663Z 
2023-06-29T16:47:58.252832663Z Please make sure you have the correct access rights
2023-06-29T16:47:58.252832663Z and the repository exists.```

File: ansible/git.yaml [hunk 3]
```diff
@@ -100,9 +148,13 @@
       KUBECONFIG: "{{ kubeconfig }}"
     register: git_secret_exist_switch
 
+  - name: creat git_url
+    set_fact:
+      git_url: "ssh://{{ git.user }}@{{ ansible_default_ipv4.address }}:{{ git.port if git.port != '22' else '' }}/home/git/playbooks.git"
```
Comment (olliewalsh) [inline ansible/git.yaml:153]: that's the culprit

File: ansible/git.yaml [hunk 4]
```diff
@@ -100,9 +148,13 @@
       KUBECONFIG: "{{ kubeconfig }}"
     register: git_secret_exist_switch
 
+  - name: creat git_url
+    set_fact:
+      git_url: "ssh://{{ git.user }}@{{ ansible_default_ipv4.address }}:{{ git.port if git.port != '22' else '' }}/home/{{ git.user }}/playbooks.git"
```
Comment (olliewalsh) [inline ansible/git.yaml:153]: need to omit : if not setting port

File: ansible/git.yaml [hunk 5]
```diff
@@ -3,46 +3,96 @@
...
         mode: '0700'
-        owner: git
-        group: git
+        owner: "{{ git.user }}"
+        group: "{{ git.group }}"
 
     - name: create playbooks git repo dir
       file:
-        path: /home/git/playbooks.git
+        path: "/home/{{ git.user }}/playbooks.git"
```
Comment (olliewalsh) [inline ansible/git.yaml:87]: Need to remove this in openstack_cleanup

**Comment by stuggi** [comment]
```
TASK [Download RHEL/CentOS guest image http://download.devel.redhat.com/brewroot/packages/rhel-guest-image/8.4/992/images/rhel-guest-image-8.4-992.x86_64.qcow2] ***
Friday 30 June 2023  01:04:12 +0000 (0:00:00.310)       0:00:07.456 *********** 
[0;31mfatal: [localhost]: FAILED! => {[0m
[0;31m    "changed": false,[0m
[0;31m    "dest": "/opt/http_store/data/images/rhel-guest-image-8.4-992.x86_64.qcow2.tmp",[0m
[0;31m    "elapsed": 0,[0m
[0;31m    "response": "HTTP Error 404: Not Found",[0m
[0;31m    "status_code": 404,[0m
[0;31m    "url": "http://download.devel.redhat.com/brewroot/packages/rhel-guest-image/8.4/992/images/rhel-guest-image-8.4-992.x86_64.qcow2"[0m
```

**Comment by abays** [comment]
> ```
> TASK [Download RHEL/CentOS guest image http://.../packages/rhel-guest-image/8.4/992/images/rhel-guest-image-8.4-992.x86_64.qcow2] ***
> Friday 30 June 2023  01:04:12 +0000 (0:00:00.310)       0:00:07.456 *********** 
> �[0;31mfatal: [localhost]: FAILED! => {�[0m
> �[0;31m    "changed": false,�[0m
> �[0;31m    "dest": "/opt/http_store/data/images/rhel-guest-image-8.4-992.x86_64.qcow2.tmp",�[0m
> �[0;31m    "elapsed": 0,�[0m
> �[0;31m    "response": "HTTP Error 404: Not Found",�[0m
> �[0;31m    "status_code": 404,�[0m
> �[0;31m    "url": "http://.../packages/rhel-guest-image/8.4/992/images/rhel-guest-image-8.4-992.x86_64.qcow2"�[0m
> ```

It (`http://download.devel.redhat.com/brewroot/packages/rhel-guest-image/8.4/992/images/rhel-guest-image-8.4-992.x86_64.qcow2`) indeed seems to be gone now.

**Pull request merged**